model/project_ref.go

```diff
@@ -794,9 +804,40 @@ func (p *ProjectRef) GetActivationTime(variant *BuildVariant) (time.Time, error)
 			continue
 		}
 
-		return buildStatus.ActivateAt.Add(time.Minute * time.Duration(p.getBatchTime(variant))), nil
+		return buildStatus.ActivateAt.Add(time.Minute * time.Duration(p.getBatchTimeForVariant(variant))), nil
+	}
+
+	return defaultRes, nil
+}
+
+func (p *ProjectRef) GetActivationTimeForTask(t *BuildVariantTaskUnit) (time.Time, error) {
+	defaultRes := time.Now()
+	if t == nil {
+		return defaultRes, nil
+	}
+	if t.CronBatchTime != "" {
+		return GetActivationTimeWithCron(time.Now(), t.CronBatchTime)
 	}
 
+	lastActivated, err := VersionFindOne(VersionByLastTaskActivation(p.Identifier, t.Variant, t.Name).WithFields(VersionBuildVariantsKey))
```

Comment:
Well, I did it similarly to how build variant activation happens, which is also by version. That being said, the new fields already have to be added to the version document since ActivateElapsedBuildsAndTasks iterates through the metadata stored in the version (otherwise we'd be calculating the activation times in that function, which would be intense).

The only field that I think doesn't translate to the task collection is Ignored.